### ISSUE TITLE
pip-installed pyright hangs

### ISSUE THREAD

**Issue by ikamensh** [RobertCraigie/pyright-python#43]
Hello,

I've been previously using npm-installed pyright, and it worked. I was excited to try pip installed version, however it didn't work, not producing any useful input on --version or call w/o arguments.

## Describe the bug 
Pip installed version of pyright doesn't launch, asked to print version it only set ERRNO to 1 in 1 minute+ running time and made no output.

## To Reproduce 
in MacOs 11.6.3, Python 3.8.8,
pip install pyright==1.1.231
pyright --version -> no version outputted, ERRNO set.

(also tried on Python 3.9.6)

Expected behavior
Expected pip version of pyright to behave same as npm version.

## Code
```
~ $ pip install pyright                                          (base) 08:08:37
...
Installing collected packages: nodeenv, pyright
Successfully installed nodeenv-1.6.0 pyright-1.1.231
~ $ pyright --version                                            (base) 08:08:45
~ [1] $ pyright --version                                        (base) 08:10:01
^CTraceback (most recent call last):
  File "~/opt/anaconda3/bin/pyright", line 8, in <module>
    sys.exit(entrypoint())
  File "~/opt/anaconda3/lib/python3.8/site-packages/pyright/cli.py", line 72, in entrypoint
    sys.exit(main(sys.argv[1:]))
  File "~/opt/anaconda3/lib/python3.8/site-packages/pyright/cli.py", line 20, in main
    return run(*args, **kwargs).returncode
  File "~/opt/anaconda3/lib/python3.8/site-packages/pyright/cli.py", line 48, in run
    return node.run('npx', *pre_args, f'pyright@{version}', *args, **kwargs)
  File "~/opt/anaconda3/lib/python3.8/site-packages/pyright/node.py", line 112, in run
    return subprocess.run(node_args, env=env, **kwargs)
  File "~/opt/anaconda3/lib/python3.8/subprocess.py", line 495, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "~/opt/anaconda3/lib/python3.8/subprocess.py", line 1020, in communicate
    self.wait()
  File "~/opt/anaconda3/lib/python3.8/subprocess.py", line 1083, in wait
    return self._wait(timeout=timeout)
  File "~/opt/anaconda3/lib/python3.8/subprocess.py", line 1808, in _wait
    (pid, sts) = self._try_wait(0)
  File "~/opt/anaconda3/lib/python3.8/subprocess.py", line 1766, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
KeyboardInterrupt
```

**Comment by RobertCraigie** [comment]
Hi @ikamensh, could you please set the environment variable `PYRIGHT_PYTHON_DEBUG` to `1` and send the additional logs, thank you!

It looks like it's hanging on the npx call, can you successfully run `npx pyright --version`?

**Comment by ikamensh** [comment]
Adding the DEBUG variable didn't help it produce any output. Where are the log files saved?

However, what made a difference was switching off company VPN with a proxy that might be blocking some URLs. Without it that worked with following output:
```
~/.c/pip [127] $ pyright --version                      (experiments_2) 20:13:58
pyright 1.1.231
npm notice
npm notice New minor version of npm available! 8.0.0 -> 8.5.5
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.5.5
npm notice Run npm install -g npm@8.5.5 to update!
npm notice
```

After once running without VPN, it now runs with VPN but apparently some web call first has to time out as it takes 1+ minute to print version.


### npx
```
(base) kai7rng@RNG-C-000K8 pip % npx pyright --version
pyright 1.1.222
```
That's the version installed with npm. It works with and without the proxy.

**Comment by RobertCraigie** [comment]
> Adding the DEBUG variable didn't help it produce any output. Where are the log files saved?

Hmmm, based on the original stack trace the logs should be output before the process hangs.

> what made a difference was switching off company VPN with a proxy that might be blocking some URLs

Yeah this seems like the culprit here, does your company VPN block npm?

> After once running without VPN, it now runs with VPN but apparently some web call first has to time out as it takes 1+ minute to print version.

That is weird, the only request that the python wrapper makes in this situation is to the PyPi API to check if you're on the latest version. This shouldn't be blocked on your company VPN as you've managed to install the package successfully, this request also has a timeout of 1s so I have no idea what is blocking here. It could be a request made by `npx`.

> That's the version installed with npm. It works with and without the proxy.

`npx` might do some weird things with and without the version specifier. Does running `npx pyright@1.1.230 --version` under the VPN work?

**Comment by ikamensh** [comment]
`npx pyright@1.1.231 --version` (the one installed with pip) hangs when behind the proxy / VPN.

While npm is not fully blocked behind the proxy / VPN, some of it's functionality isn't working as expected. I'm not working closely with node / js so can't say much more.

**Comment by RobertCraigie** [comment]
@ikamensh Thanks for the info.

Do you have any suggestions for how `pyright-python` could solve this issue? As the root cause of this issue is a company VPN which we would not have control over I cannot see how we could solve this.

**Comment by ikamensh** [comment]
I think that a CLI tool should work even offline, things like checking for updates are bells and whistles and are not worth constraining the main functionality - using pyright.

**Comment by RobertCraigie** [comment]
Yes I agree. Although you would need to run the CLI once first so that:

- `nodeenv` is cached if required
- The NPM version of pyright can be installed (I'm wary of directly bundling pyright due to licensing concerns)

Aside from these, the only feature that goes against offline support is automatically checking for updates. This feature is optional in that if you are offline then the check will timeout after 1 second and will be ignored. While this 1 second delay is not ideal, it works for now.

So there is nothing directly in pyright-python that would mean offline usage could not be supported. It seems to be that `npx` is making requests itself even after the pyright NPM package has been installed once. Looks like we would have to move away from `npx` and directly install pyright using `npm` to a cached directory.

Closing in favour of #2.
